File: src/leetcode/python/JudgeRouteCircle_657.py
class JudgeRouteCircle_657(object):
    def judgeCircle(self, moves):
        """
        :type moves: str
        :rtype: bool
        """
        circle = 0
        for m in moves:
            if m == 'U': circle += 1
            elif m == 'D': circle -= 1
            elif m == 'L': circle += 2
            elif m == 'R': circle -= 2
        return circle == 0

File: src/leetcode/java/JudgeRouteCircle_657.java
package leetcode.java;

public class JudgeRouteCircle_657 {
    public boolean judgeCircle(String moves) {
        int isCircle = 0;
        for (char m : moves.toCharArray()) {
            if (m == 'U') isCircle++;
            else if (m == 'L') isCircle += 2;
            else if (m == 'D') isCircle--;
            else if (m == 'R') isCircle -= 2;
        }
        return isCircle == 0;
    }
}
